order-book/src/test/java/com/market/banica/order/book/service/grpc/componentTests/OrderBookComponentIT.java

```diff
@@ -83,31 +96,181 @@
+        //Arrange
+        int numberOfTickResponses = 10;
+        ReflectionTestUtils.setField(orderBookService, AURORA_CLIENT, auroraClient);
+
+        String serverNameMarket = InProcessServerBuilder.generateName();
+
+        when(auroraClient.getAsynchronousStub()).thenReturn(asynchronousStub);
+
+        ManagedChannel marketChannel = InProcessChannelBuilder.forName(serverNameMarket)
+                .defaultServiceConfig(ChannelRPCConfig.getInstance().getServiceConfig())
+                .enableRetry().maxRetryAttempts(MAX_RETRY_ATTEMPTS).directExecutor().build();
+
+        // Start Market
+        startMarketService(numberOfTickResponses, serverNameMarket, THREAD_SLEEP_TIME_MARKET);
+
+        // Start OrderBook
+        startOrderBookService(orderBookService, THREAD_SLEEP_TIME_DEFAULT);
+
+        List<TickResponse> tickResponses = new ArrayList<>();
+
+        // Start Aurora
+        startAuroraService(marketChannel, tickResponses);
+
+        InterestsRequest interestsRequest = InterestsRequest.newBuilder().setClientId(clientId).setItemName(EGGS_GOOD).build();
+
+
+        //Act
+        InterestsResponse interestsResponse = blockingStub.announceItemInterest(interestsRequest);
+
+        //Assert
+        assertTrue(interestsResponse.isInitialized());
+        assertEquals(numberOfTickResponses, tickResponses.size());
+    }
+
+    @Test
+    public void auroraServiceToOrderBookRequestsRetryExecutesWithSuccess() {
+        //Arrange
+        ItemOrderBookResponse response = generateItemOrderBookExpectedResponse();
+
+        new Thread(() -> {
+            try {
+                Thread.sleep(THREAD_SLEEP_TIME_DEFAULT);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverName).directExecutor().addService(testConfiguration.getGrpcOrderBookServiceItemLayers(response))
+                        .build().start());
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        }).start();
+
+        ItemOrderBookRequest auroraRequest = ItemOrderBookRequest.newBuilder()
+                .setItemName(productName).setClientId(clientId).setQuantity(productQuantity).build();
+
+        //Act
+        ItemOrderBookResponse auroraResponse = blockingStub.getOrderBookItemLayers(auroraRequest);
+
+        ItemOrderBookResponse expected = generateItemOrderBookExpectedResponse();
+
+        //Assert
+        assertEquals(expected, auroraResponse);
+    }
+
+    @Test
+    public void calculatorToAuroraToOrderBookRequestsRetryExecutesWithSuccess() {
+        //Arrange
+        List<OrderBookLayer> layers = new ArrayList<>();
+        layers.add(OrderBookLayer.newBuilder().setPrice(productPrice).setQuantity(productQuantity).setOrigin(Origin.AMERICA).build());
+
+        when(itemMarket.getRequestedItem(productName, productQuantity)).thenReturn(layers);
+
+        // Start OrderBook
```

Comment:
instead of writing a comment what this part of the code does, extract the code in a separate method with the name of the comment for example

Reply:
Fixed

Reply:
Fixed